commands/serve/web/__tests__/connect.test.js

```diff
@@ -0,0 +1,462 @@
+          webIntegrationId: 'someIntegrationId',
+          enigma: {
+            secure: false,
+            host: 'some.eu.tenant.pte.qlikdev.com',
+          },
+        })
+      );
+    });
+
+    test('should detect engine url and integration id from provided link', () => {
+      window.location.assign(
+        `/some-url?engine_url=wss://${authConfig.host}&qlik-web-integration-id=${authConfig.webIntegrationId}`
+      );
+      expect(getParams()).toEqual({
+        engine_url: `wss://${authConfig.host}`,
+        'qlik-web-integration-id': authConfig.webIntegrationId,
+      });
+    });
+
+    test('should separate columns in to an array in case of any `cols` params in link', () => {
+      window.location.assign('/?engine_url=ws://localhost:1234&cols=col_01,col_02,col_03');
+      expect(getParams()).toEqual({
+        engine_url: 'ws://localhost:1234',
+        cols: ['col_01', 'col_02', 'col_03'],
+      });
+    });
+  });
+
+  describe('parseEngineURL()', () => {
+    let url;
+
+    test('should return invalid object if there was no match', () => {
+      url = '/?someParam=https://localhost:1234';
+      const result = parseEngineURL(url);
+      expect(result).toEqual({
+        engineUrl: url,
+        invalid: true,
+      });
+    });
+
+    test('should return expected info from SDE url', () => {
+      url = `/some-url?engine_url=wss://${authConfig.host}&qlik-web-integration-id=${authConfig.webIntegrationId}`;
+      const result = parseEngineURL(url);
+      expect(result).toMatchObject({
+        enigma: expect.objectContaining({
+          secure: expect.any(Boolean),
+          host: expect.any(String),
+          port: undefined, // because of providing a link
+          appId: undefined, // since there is not appid in link
+        }),
+        engineUrl: expect.any(String),
+        appUrl: undefined, // because no app has been provided
+      });
+    });
+
+    test('should return expected info from Local Docker url', () => {
+      url = '/?engine_url=ws://localhost:1234';
+      const result = parseEngineURL(url);
+      expect(result).toMatchObject({
+        enigma: expect.objectContaining({
+          secure: expect.any(Boolean),
+          host: expect.any(String),
+          port: expect.any(String),
+          appId: undefined, // since there is not appid in link
+        }),
+        engineUrl: expect.any(String),
+        appUrl: undefined, // because no app has been provided
+      });
+    });
+
+    test('should find app url param from provided link', () => {
+      url = `/app/SOME_APP_ID/?engine_url=wss://${authConfig.host}`;
+      const result = parseEngineURL(url);
+      console.log({ result });
+      expect(result).toMatchObject({
+        enigma: expect.objectContaining({
+          secure: expect.any(Boolean),
+          host: expect.any(String),
+          port: undefined, // because of providing a link
+          appId: 'SOME_APP_ID/', // since there is not appid in link
```

Comment:
I've noticed that when we fall in here: https://github.com/qlik-oss/nebula.js/blob/5990d85a95772b5e979efb80b29ef3c22eb7e453/commands/serve/web/connect.js#L47

we keep the `/` at the end of the extracted `appId` variable. Is it ok to have it like that?